src/algebra/group/pi.lean

```diff
@@ -182,3 +182,25 @@ lemma pi.single_mul [Π i, monoid_with_zero $ f i] (i : I) (x y : f i) :
 (mul_hom.single f i).map_mul x y
 
 end single
+
+section piecewise
+
+@[to_additive]
+lemma pi.piecewise_mul [Π i, has_mul (f i)] (s : set I) [Π i, decidable (i ∈ s)]
+  (f₁ f₂ g₁ g₂ : Π i, f i) :
+  s.piecewise (f₁ * f₂) (g₁ * g₂) = s.piecewise f₁ g₁ * s.piecewise f₂ g₂ :=
+by { ext i, by_cases hi : i ∈ s; simp [*, set.piecewise] }
+
+@[to_additive]
+lemma pi.piecewise_inv [Π i, has_inv (f i)] (s : set I) [Π i, decidable (i ∈ s)]
+  (f₁ g₁ : Π i, f i) :
+  s.piecewise (f₁⁻¹) (g₁⁻¹) = (s.piecewise f₁ g₁)⁻¹ :=
+by { ext i, by_cases hi : i ∈ s; simp [*, set.piecewise] }
+
+@[to_additive]
+lemma pi.piecewise_div [Π i, has_div (f i)] (s : set I) [Π i, decidable (i ∈ s)]
+  (f₁ f₂ g₁ g₂ : Π i, f i) :
+  s.piecewise (f₁ / f₂) (g₁ / g₂) = s.piecewise f₁ g₁ / s.piecewise f₂ g₂ :=
+by { ext i, by_cases hi : i ∈ s; simp [*, set.piecewise] }
```

Comment:
These lemmas generalize:
```lean
/-- A fiber-wise version of `set.comp_piecewise` -/
lemma set.comp_piecewise' (s : set I) [Π i, decidable (i ∈ s)]
  (f₁ f₂ : Π i, f i) (h₁ : Π i, f i → g i) (i) :
  h₁ i (s.piecewise f₁ f₂ i) = s.piecewise (λ i, h₁ i (f₁ i)) (λ i, h₁ i (f₂ i)) i :=
by { by_cases hi : i ∈ s; simp [*, set.piecewise],}
```
```lean
@[to_additive]
lemma pi.piecewise_inv [Π i, has_inv (f i)] (s : set I) [Π i, decidable (i ∈ s)]
  (f₁ g₁ : Π i, f i) :
  s.piecewise (f₁⁻¹) (g₁⁻¹) = (s.piecewise f₁ g₁)⁻¹ :=
(funext $ set.comp_piecewise' _ _ s f₁ g₁ (λ i, has_inv.inv)).symm
```
and a similar approach for

Reply:
This file doesn't import `data/set/function`.

Reply:
Where am I using `data/set/function`?

Reply:
I mean, general lemmas about `piecewise` like `set.comp_piecewise'` should go to `data.set.function`. Probably we should just import this file.

Reply:
`import`ed `data.set.function`, added generic lemmas (similarly to `pi.single_op`).